x-pack/plugins/alerts/server/types.ts

```diff
@@ -126,13 +126,31 @@ export interface RawAlert extends SavedObjectAttributes {
   createdBy: string | null;
   updatedBy: string | null;
   createdAt: string;
-  apiKey: string | null;
-  apiKeyOwner: string | null;
+  apiKey?: string | null;
+  apiKeyOwner?: string | null;
   throttle: string | null;
   muteAll: boolean;
   mutedInstanceIds: string[];
 }
 
+export type AlertUpdateRequiredFields = Pick<
+  RawAlert,
+  | 'enabled'
+  | 'name'
+  | 'tags'
+  | 'alertTypeId'
+  | 'consumer'
+  | 'schedule'
+  | 'actions'
+  | 'params'
+  | 'createdBy'
+  | 'updatedBy'
+  | 'createdAt'
+  | 'throttle'
+  | 'muteAll'
+  | 'mutedInstanceIds'
+>;
+
```

Comment:
Sadly, using `Omit` doesn't work because extending `SavedObjectAttributes` has confused Typescript and it can't figure out what the inverse is - so we have to d oit manually.